### PR TITLE
fix: avoid indefinite spinny wheel after completing kyc with Persona

### PR DESCRIPTION
**Pull request by cajubelt**
also added test for this

### Description

There is a bug where after completing KYC with Persona, returning to the link bank account steps screen results in a spinny wheel indefinitely over step one. This fixes the bug by removing the spinny wheel with an onSuccess callback. 

### Other changes

na

### Tested

added unit test and tested manually

### How others should test

- with an account with a verified phone number:
- navigate to settings
- tap 'link bank account'
- tap 'begin' under Step 1
- go through Persona flow
- when complete, returning to the link bank account steps screen should show step 1 as completed (rather than a spinny wheel that never goes away)

### Related issues

na

### Backwards compatibility

backwards compatible

### CODE REVIEW

File: packages/mobile/src/account/LinkBankAccountScreen.test.tsx [hunk 1]
```diff
@@ -84,4 +97,30 @@ describe('LinkBankAccountScreen', () => {
...
+      const { getByText, getByTestId, rerender } = render(<StepOne kycStatus={undefined} />)
+      await waitFor(() => expect(getByTestId('PersonaButton')).not.toBeDisabled())
+
+      await fireEvent.press(getByTestId('PersonaButton'))
+      await waitFor(() => getByText('linkBankAccountScreen.verifying.title'))
+
+      rerender(<StepOne kycStatus={KycStatus.Approved} />) // simulating the redux store being updated to reflect new KYC status. in a production flow Persona calls an IHL webhook, which updates firebase, which triggers a redux store update
+
+      expect(personaButtonSuccessCallback).toBeTruthy()
+      personaButtonSuccessCallback?.()
```
Comment (jh2oman) [inline packages/mobile/src/account/LinkBankAccountScreen.test.tsx:122]:
```suggestion
      personaButtonSuccessCallback?.() // simulating the persona onSuccess hook being called when the user completes KYC
```

**Review by jh2oman — APPROVED** [review]
These look great! Sometimes these components can be very difficult to test, but I think this ones done well. That little `MockPersona` really expands the testing ability

File: packages/mobile/src/account/Persona.test.tsx [hunk 2]
```diff
@@ -74,5 +104,10 @@ describe('Persona', () => {
 
     fireEvent.press(getByTestId('PersonaButton'))
     expect(Inquiry.fromTemplate).toHaveBeenCalledWith(FAKE_TEMPLATE_ID)
+
+    expect(mockInquiryBuilder.onSuccess).toHaveBeenCalled()
+    expect(personaProps.onSuccess).not.toHaveBeenCalled()
+    mockInquiryBuilder.onSuccess.mock.calls?.[0]?.[0]?.()
```
Comment (jh2oman) [inline packages/mobile/src/account/Persona.test.tsx:110]: I don't really follow whats going on here. Can you explain and add a comment?

**Review by jh2oman — APPROVED** [review]

**Comment by ValoraQA** [comment]
Hey @MuckT We have verified the above task in Latest iOS Alfajores build and Latest Android Alfajores build V:1.25.1 and updated the status in test rails 

**iOS 14 :**

**Android 11:**

**Observation :** Few components are not seen on Android application as compared with iOS and we have raised a bug for it 

[Bug link](https://github.com/valora-inc/wallet/issues/1915)

**Devices :** iPhone 12 (14.7.1), Google Pixel 2XL (11.0)

**CC:** @cajubelt 

Thanks:.!

**Comment by ValoraQA** [comment]
Hey @MuckT We have verified the above task in Latest Android Alfajores build V:1.25.2 and updated the status in test rails

**Android 10:**
https://valoraapp.testrail.io/index.php?/tests/view/7073&group_by=cases:section_id&group_order=asc&group_id=30

**Observation** : 
1. Few components are not seen on Android application as compared with iOS and we have raised a [bug](https://github.com/valora-inc/wallet/issues/1915) for it

2."Get Access" is seen for link bank account section even after the Phone number is already verified after user restores the account and we have raised a [bug](https://github.com/valora-inc/wallet/issues/1886) for it.

**Devices** : Samsung Galaxy M40(10.0)

**CC:** @cajubelt 

Thanks:.!

**Comment by ValoraQA** [comment]
Hey @MuckT, We have verified the above task in Latest Test flight Alfajores build V:1.25.2 and updated the status in test rails

**Test rail link:**
**iOS 15:** https://valoraapp.testrail.io/index.php?/tests/view/4455
 
Bug: [Link](https://github.com/valora-inc/wallet/issues/1886)

**Devices :** iPhone 13 mini (15.1.1)
CC: @cajubelt 
Thanks:.!!

**Comment by ValoraQA** [comment]
Hey @MuckT We have verified the above task in Latest Android Alfajores build V:1.25.2 and updated the status in test rails

**Android 8:**
https://valoraapp.testrail.io/index.php?/tests/view/7136&group_by=cases:section_id&group_order=asc&group_id=30

**Observation :**

Few components are not seen on Android application as compared with iOS and we have raised a [bug](https://github.com/valora-inc/wallet/issues/1915) for it.

"Get Access" is seen for link bank account section even after the Phone number is already verified after user restores the account and we have raised a [bug](https://github.com/valora-inc/wallet/issues/1886) for it.

Devices : Samsung Galaxy J5 Prime (8.0.0)

CC: @cajubelt

Thanks:.!

**Comment by ValoraQA** [comment]
Hey @MuckT, We have verified the above task in Latest Android Alfajores build V:1.25.2 and updated the status in test rails

**Test rail link:**
**Android 9:** https://valoraapp.testrail.io/index.php?/tests/view/7460

**Observation :**
- Few components are not seen on Android application as compared with iOS and we have raised a [bug](https://github.com/valora-inc/wallet/issues/1915) for it.

- "Get Access" is seen for link bank account section even after the Phone number is already verified after user restores the account and we have raised a [bug](https://github.com/valora-inc/wallet/issues/1886) for it.

**Device:** Redmi Note 5 Pro (9.0)
cc: @cajubelt 
Thanks..!!

**Pull request merged**